server/data/nonAssociationsApi.ts

```diff
@@ -1,466 +1,38 @@
-      return parseDates(nonAssociation)
-    })
-  }
-
-  /**
-   * Create a new non-association
-   *
-   * @throws SanitisedError<ErrorResponse>
-   */
-  createNonAssociation(request: CreateNonAssociationRequest): Promise<OpenNonAssociation> {
-    return this.post<OpenNonAssociation>({
-      path: '/non-associations',
-      data: request as unknown as Record<string, unknown>,
-    }).then(parseDates)
-  }
-
-  /**
-   * Update an existing new non-association by ID
-   *
-   * @throws SanitisedError<ErrorResponse>
-   */
-  updateNonAssociation(id: number, request: UpdateNonAssociationRequest): Promise<NonAssociation> {
-    return this.patch<NonAssociation>({
-      path: `/non-associations/${encodeURIComponent(id)}`,
-      data: request as unknown as Record<string, unknown>,
-    }).then(parseDates)
-  }
-
-  /**
-   * Close an open non-association by ID
-   *
-   * @throws SanitisedError<ErrorResponse>
-   */
-  closeNonAssociation(id: number, request: CloseNonAssociationRequest): Promise<ClosedNonAssociation> {
-    return this.put<ClosedNonAssociation>({
-      path: `/non-associations/${encodeURIComponent(id)}/close`,
-      data: request as unknown as Record<string, unknown>,
-    }).then(parseDates)
+    super(systemToken, config.apis.hmppsNonAssociationsApi, logger, [restClientMetricsMiddleware])
   }
 }
 
-export function parseDates<O extends { whenCreated: unknown; whenUpdated: unknown; closedAt: unknown }>(data: O): O {
-  // eslint-disable-next-line no-param-reassign
-  data.whenCreated = new Date(data.whenCreated as string)
-  // eslint-disable-next-line no-param-reassign
-  data.whenUpdated = new Date(data.whenUpdated as string)
-  if (data.closedAt) {
-    // eslint-disable-next-line no-param-reassign
-    data.closedAt = new Date(data.closedAt as string)
+const systemUsers: ReadonlyArray<StaffMember> = systemUsernames.map(username => {
+  return {
+    username,
+    firstName: 'System',
+    lastName: '',
   }
-  return data
-}
-
-// known system users should appear here:
-const systemUsers: ReadonlyArray<StaffMember> = [
-  // https://github.com/ministryofjustice/hmpps-non-associations-api/blob/04bf15fd1a7d659abe785749fbedda9f13627fba/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/HmppsNonAssociationsApi.kt#L9
-  { username: 'NON_ASSOCIATIONS_API', firstName: 'System', lastName: '' },
-  { username: 'PRISONER_MANAGER_API', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-from-nomis-migration-non-associations', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-from-nomis-migration-non-associations-1', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-to-nomis-update-non-associations', firstName: 'System', lastName: '' },
-  { username: 'hmpps-prisoner-to-nomis-update-non-associations-1', firstName: 'System', lastName: '' },
-]
+})
 
 /**
```

Comment:
dealing with staff user lookups and non-association grouping is specific to this application and was not extracted into the shared library